src/server/v2.0/handler/base.go

```diff
@@ -95,3 +98,32 @@ func (b *BaseAPI) RequireProjectAccess(ctx context.Context, projectIDOrName inte
 	}
 	return ierror.ForbiddenError(nil)
 }
+
+// Links return Links based on the provided pagination information
+// The parameter url is value rather the reference to avoid overwrite the original values
+func (b *BaseAPI) Links(ctx context.Context, url url.URL, total, pageNumber, pageSize int64) internal.Links {
```

Comment:
`url` parameter can be a pointer here so that every where to call this method not needed to covert the URL of the http.Request t to non pointer.

Reply:
As the comment of the method said: `The parameter url is value rather the reference to avoid overwrite the original values`

Reply:
Why not copy the URL in the method?